src/main.rs

```diff
@@ -57,6 +66,28 @@ impl Backend {
             )
             .await;
     }
+
+    async fn run_fixing_and_report_fixes(&self, uri: &Url) {
+        let per_file_state = self.per_file.get(uri).unwrap();
+        let mut cloned_contents = per_file_state.contents.clone();
```

Comment:
We need to diff against the original contents so clone it (cloning a `Rope` is apparently cheap) and then let `run_fixing_for_slice()` mutate the clone